File: geoservice-isochrone/src/main/java/com/github/cunvoas/geoserviceisochrone/controller/mobile/auth/dto/LoginRequest.java
package com.github.cunvoas.geoserviceisochrone.controller.mobile.auth.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.Data;

/**
 * DTO pour la requête de login.
 */
@Data
public class LoginRequest {

    @NotBlank(message = "Le nom d'utilisateur est requis")
    private String username;

    @NotBlank(message = "Le mot de passe est requis")
    private String password;
}

File: geoservice-mobile/lib/models/login_request.dart
/// Modèle pour la requête de login
class LoginRequest {
  final String username;
  final String password;

  LoginRequest({
    required this.username,
    required this.password,
  });

  /// Convertit l'instance en JSON
  Map<String, dynamic> toJson() {
    return {
      'username': username,
      'password': password,
    };
  }
}
